### PR TITLE
add model data ingestion proto

### PR DESCRIPTION
**Pull request by sunb26**
* model name enum based on api model name. If further version suffix needed - I can manually find and append the release date of each model

This means I will need the lab folks to specify the version and date the experiments were conducted because some models like the microsoft ones may have been changed since even for the same model.

### CODE REVIEW

File: pb/lmsysmd/load/v1/model.proto [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+message Model {
+  ModelId model_id = 1 [ 
+    (buf.validate.field).required = true, 
+    (buf.validate.field).enum.defined_only = true 
+  ];
+  string display_name = 2 [ 
+    (buf.validate.field).required = true, 
+    (buf.validate.field).string.min_len = 2,
+    (buf.validate.field).string.max_len = 20
```
Comment (zackxzr) [inline pb/lmsysmd/load/v1/model.proto:32]: Might need a larger buffer here. Some models can exceed this length, especially if we are including the release date.

File: pb/lmsysmd/load/v1/model.proto [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  ];
+  google.protobuf.Timestamp release_time = 6 [ 
+    (buf.validate.field).required = true,
+    (buf.validate.field).timestamp.lt_now = true,
+    (buf.validate.field).timestamp.gt = { seconds: 1577836800 }
+  ]; // release_time must be after 2020-01-01 and before now
+}
+
+enum ModelId {
+  MODEL_ID_UNSPECIFIED = 0;
```
[the remark this answers is not in the document]
Comment (sunb26) [reply pb/lmsysmd/load/v1/model.proto:58]: This is per AIP guidelines unfortunately :( 

https://google.aip.dev/126

File: pb/lmsysmd/load/v1/model.proto [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  MODEL_ID_LLAMA_3_8B = 11;
+  MODEL_ID_PHI_3_MEDIUM = 12;
+  MODEL_ID_CLAUDE_3_HAIKU_20240307 = 13;
+  MODEL_ID_CLAUDE_3_SONNET_20240229 = 14;
+  MODEL_ID_CLAUDE_3_5_SONNET_20240620 = 15;
+  MODEL_ID_YI_LARGE = 16;
+}
+
+enum ModelProvider {
+  MODEL_PROVIDER_UNSPECIFIED = 0;
```
[the remark this answers is not in the document]
Comment (sunb26) [reply pb/lmsysmd/load/v1/model.proto:78]: This is based on the current spreadsheet provided 

https://docs.google.com/spreadsheets/d/1-skT0JRJREJfn5qIQtSwyz2KVHMfLPJy0hOBfjyggZA/edit?gid=0#gid=0

to add new models and providers we will have to manually add them to the enum but it is infrequent so this is feasible

File: pb/lmsysmd/load/v1/model.proto [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    option idempotency_level = IDEMPOTENT; 
+  }
+}
+
+message BatchCreateModelRequest {
+  string google_sheet_url = 1 [ 
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.uri = true,
+    (buf.validate.field).string.min_len = 2,
+    (buf.validate.field).string.max_len = 60
```
[the remark this answers is not in the document]
Comment (sunb26) [reply pb/lmsysmd/load/v1/model.proto:20]: Yeah you're probably right I'll up it

File: pb/lmsysmd/load/v1/model.proto [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  string model_provider_url = 5 [ 
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.uri = true,
+    (buf.validate.field).string.min_len = 2,
+    (buf.validate.field).string.max_len = 150
+  ];
+  google.protobuf.Timestamp release_time = 6 [ 
+    (buf.validate.field).required = true,
+    (buf.validate.field).timestamp.lt_now = true,
+    (buf.validate.field).timestamp.gt = { seconds: 1577836800 }
```
Comment (Lev1ty) [inline pb/lmsysmd/load/v1/model.proto:53]:
```suggestion
    (buf.validate.field).timestamp.gt.seconds = 1577836800
```

**Review by zackxzr — APPROVED** [review]

**Review by Lev1ty — APPROVED** [review]

**Comment by Lev1ty** [comment]
please fix lints before merging :)

**Pull request merged**